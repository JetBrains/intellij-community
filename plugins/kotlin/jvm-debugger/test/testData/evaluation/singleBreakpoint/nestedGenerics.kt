class A<T>(x : T) {
    val prop: Pair<Pair<T, T>, T> = (x to x) to x

    fun func(): Int{
        //Breakpoint!
        return 42
    }
}

fun main() {
    A("").func()
}

// EXPRESSION: prop
// RESULT: instance of kotlin.Pair(id=ID): Lkotlin/Pair;