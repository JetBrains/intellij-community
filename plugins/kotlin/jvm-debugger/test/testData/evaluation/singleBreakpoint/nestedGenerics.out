LineBreakpoint created at nestedGenerics.kt:6
Run Java
Connected to the target VM
nestedGenerics.kt:6
Compile bytecode for prop
Disconnected from the target VM

Process finished with exit code 0
